userspace/libscap/scap_reader.h

```diff
@@ -0,0 +1,199 @@
+/*
```

Comment:
Same. It feels like nobody except libscap/engine/savefile should care about the existence of a scap_reader

Reply:
In `scap-int.h` we have two functions using that type: `scap_read_init` and `scap_fd_read_from_disk`. Now, those are at the libscap/ level but only used by the savefile engine too. I'll try to move all this inside the savefile engine if you're ok with it.

Reply:
Sure. kill_it_with_fire.jpg

I feel like this is a good opportunity to reduce the API surface of libscap, even if just a little. Many of the functions in scap-int.h are there just because they're defined in private component A and used in private component B, but effectively become public as it's hard *not* to include scap-int.h from a consumer at one point or another.

Reply:
I was able to push `scap_reader` into the savefile engine, but zlib definitions are still required at the base level directory due to `scap_dumper` using it. That's part of the public API too, so I what do you think about keeping it there for now?

Reply:
Yeah, let's not turn this PR into another rewrite-the-world exercise